### ISSUE TITLE
NaN loss when training with sine

### ISSUE THREAD

**Issue by sidml** [cnguyen10/few_shot_meta_learning#2]
Hi.
Thanks you so much for sharing the code.
i cloned your repo and i tried to run abml.py with sine curve. I get nan loss and the code exits.
Please let me know if i need to change any hyper parameters for this task.
Thanks.

**Comment by cnguyen10** [comment]
Hi @sidml 
Thank you for your interest in the code. I run the abml.py again today, but did not have any errors as you reported. Could you please provide more information about the failure? Did you use the same setting and same network architecture?

**Comment by sidml** [comment]
Thanks for your reply.
When i run abml.py, it exits after giving nan.

Here's the output.
![image](https://user-images.githubusercontent.com/51043033/78859314-8c26f080-7a61-11ea-825a-c094bb4c8389.png)

I just changed this line to use sine. Everything else is exactly the same.
`parser.add_argument(
    '--datasource',
    type=str,
    default='sine_line',
    help='\'omniglot\' or \'miniImageNet\' or \'sine_line\''
)`

**Comment by cnguyen10** [comment]
If only the dataset is changed to 'sine_line' and everything is kept to be default, it will be a 1-shot regression as shown in your screenshot, while in the implement, I tested in 5-shot setting. To run it under the 5-shot setting, please specify '--k_shot=5' when calling abml.py (or you can simply copy the first comment at the beginning of the abml.py file).
The reason you got NaN for 1-shot regression might be overfitting, leading to a very large value of loss at the beginning of the training, and consequently, resulting in an overflow. My very first thought  to work around would be reduced the meta-learning rate. You can also anneal the kl_reweight starting from 0 and graduately increase to 1.

**Comment by sidml** [comment]
Thanks. It works when k_shot = 5.
